src/ar.erl

```diff
@@ -248,8 +262,11 @@ start(
 	if Clean -> ar_storage:clear(); true -> do_nothing end,
 	%% Register prometheus stats collector.
 	application:ensure_started(prometheus),
+	application:ensure_all_started(prometheus_cowboy),
 	prometheus_registry:register_collector(prometheus_process_collector),
 	prometheus_registry:register_collector(ar_metrics_collector),
+	%% Start Cowboy and its dependencies
+	application:ensure_all_started(cowboy),
```

Comment:
I see no reason not to pattern match on the expected returned ok tuple. We should probably do this on all the `application:ensure_all_started/1` calls in this function.

Reply:
In the `{ok, Started}` tuple, is the order of the started apps deterministic?

Reply:
`application` works out the order of starting based on the graph of dependencies, so the order might not be deterministic if there is more than one possible ordering.  I generally use something like:

```
{ok, Started} = application:ensure_all_started(cowboy).
```